### ISSUE TITLE
Introduce content components

### ISSUE THREAD

**Issue by Exelord** [react-bootstrap/react-bootstrap#5152]
## Motivation

I was going through [Grid](https://react-bootstrap.github.io/layout/grid/) docs and really like the idea of how the bootstrap API has been abstracted to those components. I opened Bootstrap docs and was wondering if we could do the same with the content components, like `Heading`, `DisplayHeading`, `List`, `Code`, `Block (div)` etc... In other words, have an abstraction on all other elements, that are "delivered" with bootstrap. This will allow unifying those elements across the project, and abstracting bootstrap API even further.

## Solution

**Introduce content components that will handle bootstrap API and expose a simpler interface.**

This is just an idea, so I didn't focus here on every component we could implement but for sure I would be in favor of implementing the following:
- Heading
- DisplayHeading
- Text (formerly p)
- Block (formerly div)
- List

What they could handle? The same what Bootstrap allows on. This could be for example a `Heading` component

```jsx
<Heading lead level={1} align="center">Title</Heading>
```

which will produce:

```jsx
<h1 class="text-center lead">Title</h1>
```

## Alternatives

Build them on your own. Use CSS classes.

## Additional context

Example:
```jsx
<Block align="center">
  <Heading level={1}>Hello!</Heading>
  <Text>These are content components</Text>
</Block>
```

**Comment by kosmiq** [comment]
I'm not sure about this myself. I'm all in favor for components that makes stuff easier and short to write (Row, Col are great examples of this, the grid is awesomely implemented and you noted as well).

I do however not think that adding components for <Heading> would actually solve real issues. Your examples produces more code rather than less and I think that a lot of the default html is best left alone, headings (h1, h2 etc) are standard enough as they are.

There might be other opinions though.

It's been partly discussed here: https://github.com/react-bootstrap/react-bootstrap/issues/3029.

**Comment by jquense** [comment]
I think UI primitives are generally a good idea. We write the equivalents of Text, Block, Heading in every app so I think they are good ideas _generally_. Which sorts of components are good fit in RB I think depends on what the API is. I also think that success is more about making good components for concepts that bootstrap provides and not creating general app primitives from scratch. Which is to say that like discussed in #3029 something that nicely encapsulates the many utility classes in some way.

**Comment by taion** [comment]
Right... I don't think we should necessarily introduce _components_ per se that aren't in Bootstrap without good reason...

**Comment by jquense** [comment]
one could argue that Text, Flex, etc are components, if we are going by the class names

**Comment by jquense** [comment]
IDK, I think starting with a Layout/Block/View component is worthwhile probably?

**Comment by taion** [comment]
sure, we certainly use our `<Layout>` component often enough :p

**Comment by Exelord** [comment]
Even if in some cases these components will produce more code, I still think they are much more readable and brings standardization to the project. In all of my projects Ive been working on, I always introduce a component for every single html element to abstract it (gain the control) and make it meaningful. That also helps to keep the code consistent and in case of rewrites/migrations its painless to continue. 

If you take a look at react-native or even react-native-web you will see their use the same pattern. Of course, the reason is more complex but it still would give us the same advantages and control over the underlying element.

Abstracting the classes to component API is, in my opinion, a good way to forget about CSS framework abstractions, syntax, and especially when dealing with content components that are used the most, it can really unify the template layers and isolate the logic.

I would definitely appreciate having these components from react-bootstrap instead of creating them on my own.

**Comment by miltonmc** [comment]
IMHO, we should abstract as much we can from the Bootstrap's API rather than use directly CSS classes to enable one of its features. It can be especially helpful those using Typescrip, and it seems we are planning to support it GH-5148. That would allow developers to discovery the component API and its possible values ​​directly from IDE (though the IntelliSense), leaving the css classes for customisations only.

In other hand, the increase on the amount of components could lead to inadvertent developers making mistakes. For instance, they may misuse components by do an improperly composition. But thats a price I definitively would afford, and we also can mitigate this with a well written documentation.

**Comment by bogdansoare** [comment]
Any news on this?
